sympy/functions/elementary/miscellaneous.py

```diff
@@ -606,24 +605,24 @@ def _is_connected(cls, x, y):
         """
         Check if x and y are connected somehow.
         """
-        def hit(v, t, f):
-            if not v.is_Relational:
-                return t if v else f
         for i in range(2):
             if x == y:
                 return True
-            r = hit(x >= y, Max, Min)
-            if r is not None:
-                return r
-            r = hit(y <= x, Max, Min)
-            if r is not None:
-                return r
-            r = hit(x <= y, Min, Max)
-            if r is not None:
-                return r
-            r = hit(y >= x, Min, Max)
-            if r is not None:
-                return r
+            t, f = Max, Min
+            for op in range(2):
+                for j in range(2):
+                    try:
+                        if op == 0:
+                            v = x >= y
+                        else:
+                            v = x <= y
+                    except TypeError:
+                        return False  # non-real arg
+                    if not v.is_Relational:
+                        return t if v else f
+                    t, f = f, t
+                    x, y = y, x
```

Comment:
It is a bit confusing swapping arguments, but at the same time leads to pretty compact code, so I guess this is OK. Maybe add a comment on the next line like `# Restore order`?